### PR TITLE
fix: Increase Zoom Limit

### PR DESCRIPTION
**Pull request by ZhengIce**
fix:
   * Increase the zoom limit to 10% to 400%
   * Fix scaling value accuracy issues

### CODE REVIEW

**Review by zyc9012 — CHANGES_REQUESTED** [review]
Let's move the limitation into `zoomTo`, and use constants in `zoom.ts` :

https://github.com/ruilisi/fortune-sheet/blob/ee31528150f310b321ff7231385937a38b3be292/packages/core/src/modules/zoom.ts#L6-L7

And what do you mean by "value accuracy issues"?

**Comment by ZhengIce** [comment]
Numerical accuracy, such as a scaling ratio of 0.1, which is actually 0.1000000001

File: packages/react/src/components/ZoomControl/index.tsx [hunk 1]
```diff
@@ -44,6 +44,9 @@ const presets = [
   },
 ];
 
+const MAX_ZOOM_RATIO = 4;
+const MIN_ZOOM_RATIO = 0.1;
```
Comment (zyc9012) [inline packages/react/src/components/ZoomControl/index.tsx:48]: It does not make any sense if you copy-paste them. You export and import.

File: packages/react/src/components/ZoomControl/index.tsx [hunk 2]
```diff
@@ -79,7 +82,9 @@ const ZoomControl: React.FC = () => {
       <div
         className="fortune-zoom-button"
         onClick={(e) => {
-          zoomTo(context.zoomRatio - 0.1);
+          if (parseFloat(context.zoomRatio.toFixed(1)) > MIN_ZOOM_RATIO) {
+            zoomTo(parseFloat(context.zoomRatio.toFixed(1)) - 0.1);
+          }
```
Comment (zyc9012) [inline packages/react/src/components/ZoomControl/index.tsx:87]: I mean, you add the condition and `parseFloat` inside `zoomTo`.

And your "Fix scaling value accuracy issues" is wrong.

```js
parseFloat(3.7.toFixed(1)) + 0.1
> 3.8000000000000003
parseFloat((3.7 + 0.1).toFixed(1))
> 3.8
```

**Review by zyc9012 — CHANGES_REQUESTED** [review]
I mean,

File: packages/react/src/components/ZoomControl/index.tsx (hunk 2 above)
Comment (ZhengIce) [reply packages/react/src/components/ZoomControl/index.tsx:87]: I understand now. Thank you.

File: packages/react/src/components/ZoomControl/index.tsx (hunk 1 above)
Comment (ZhengIce) [reply packages/react/src/components/ZoomControl/index.tsx:48]: Do you mean for me to use a single if statement to handle the limiting process alone, just like you did in the `zoom.ts` file?

Comment (zyc9012) [reply packages/react/src/components/ZoomControl/index.tsx:48]: Yeah, just check the range at the beginning of `zoomTo`

File: packages/react/src/components/ZoomControl/index.tsx (hunk 2 above)
Comment (zyc9012) [reply packages/react/src/components/ZoomControl/index.tsx:87]: Oh, and don't forget to update `zoom.ts` as well

**Pull request merged**